include/deal.II/grid/tria.h

```diff
@@ -511,6 +513,167 @@ namespace internal
 /*------------------------------------------------------------------------*/
 
 
+namespace construction
+{
+  /**
+   * Configuration flags for fully distributed Triangulations.
+   * Settings can be combined using bitwise OR.
+   *
+   * @author Peter Munch, 2019
+   */
+  enum Settings
+  {
+    /**
+     * Default settings, other options are disabled.
+     */
+    default_setting = 0x0,
+    /**
+     * This flags needs to be set to use the geometric multigrid
+     * functionality. This option requires additional computation and
+     * communication.
+     */
+    construct_multigrid_hierarchy = 0x1
+  };
+
+  /**
+   * Information needed for each locally relevant cell, stored in
+   * ConstructionData and used during construction of a
+   * parallel::fullydistributed::Triangulation. This struct stores
+   * the cell id, the subdomain_id and the level_subdomain_id as well as
+   * information related to manifold_id and boundary_id.
+   *
+   * @author Peter Munch, 2019
+   */
+  template <int dim>
+  struct CellData
+  {
+    /**
+     * Constructor.
+     */
+    CellData() = default;
+
+    /**
+     * Boost serialization function
+     */
+    template <class Archive>
+    void
+    serialize(Archive &ar, const unsigned int /*version*/);
+
+    /**
+     * Comparison operator.
+     */
+    bool
+    operator==(const CellData<dim> &other) const;
+
+    /**
+     * Unique CellID of the cell.
+     */
+    CellId::binary_type id;
+
+    /**
+     * subdomain_id of the cell.
+     */
+    types::subdomain_id subdomain_id;
+
+    /**
+     * level_subdomain_id of the cell.
+     */
+    types::subdomain_id level_subdomain_id;
+
+    /**
+     * Manifold id of the cell.
+     */
+    types::material_id manifold_id;
+
+    /**
+     * Manifold id of all lines of the cell.
+     *
+     * @note Only used for 2D and 3D.
+     */
+    std::array<types::manifold_id, GeometryInfo<dim>::lines_per_cell>
+      manifold_line_ids;
+
+    /**
+     * Manifold id of all face quads of the cell.
+     *
+     * @note Only used for 3D.
+     */
+    std::array<types::manifold_id,
+               dim == 1 ? 1 : GeometryInfo<3>::quads_per_cell>
+      manifold_quad_ids;
+
+    /**
+     * List of face number and boundary id of all non-internal faces of the
+     * cell.
+     */
+    std::vector<std::pair<unsigned int, types::boundary_id>> boundary_ids;
+  };
+
+
+
+  /**
+   * Data used in Triangulation::create_triangulation().
+   *
+   * @author Peter Munch, 2019
+   */
+  template <int dim, int spacedim>
+  struct ConstructionData
+  {
+    /**
+     * Boost serialization function
+     */
+    template <class Archive>
+    void
+    serialize(Archive &ar, const unsigned int /*version*/);
+
+    /**
+     * Comparison operator.
+     */
+    bool
+    operator==(const ConstructionData<dim, spacedim> &other) const;
+
+    /**
+     * Cells of the locally-relevant coarse-grid triangulation.
+     */
+    std::vector<dealii::CellData<dim>> coarse_cells;
```

Comment:
If you derive the new `CellData` class from `::CellData`, then you don't need the separate `cell_infos` variable below. I think that would make things substantially simpler since the two objects currently don't really have clearly separated purposes.

Reply:
The same as above. This is the coarse-grid description, where we only need the `vertices` from `::CellData`.